scripts/west_commands/build.py

```diff
@@ -130,6 +130,13 @@ def do_add_parser(self, parser_adder):
         group.add_argument('-n', '--just-print', '--dry-run', '--recon',
                             dest='dry_run', action='store_true',
                             help="just print build commands; don't run them")
+        group.add_argument('-S', '--snippet', dest='snippets',
+                           action='append', default=[],
+                           help='''add the argument to SNIPPET; may be given
+                           multiple times. Forces CMake to run again if given.
+                           Do not use this option with manually specified
+                           -DSNIPPET... cmake arguments: the results are
+                           undefined''')
```

Comment:
This is very bad implementation. I'm still in the process of understanding this whole PR, but by this comment I can say straight ahead that this is wrong. Re-running cmake? You mean cmake-configure step or build step? Can't it be done better to run it only once like it should? I'll try to propose better alternative but I still need more time to uderstand this PR content fully.
And this part:

> Do not use this option with manually specified -DSNIPPET... cmake arguments

This is really bad, espacially that adding -D flags is part of `west build` public interface to use! (-- [cmake_opt] flag). And now you specify some "do not use" rules. What if I have snippets too in my zephyr application that are something utterly different?

Reply:
cmake to run means configuring, ninja or make would be the build. It would be invalid to not re-run cmake if the shield variable has changed - how would that be propagated to the application?

Then use -S multiple times? -DSNIPPET=... is fine when configuring using cmake, when configuring use west, different rules apply